File: Luogu/ProblemSet/Volume (2300-2399)/Luogu2397/Luogu2397-Solution-01.go
// TODO: MLE

package main

import (
	"bufio"
	"fmt"
	"os"
)

func solve(in *bufio.Reader, out *bufio.Writer) {
	var n int
	fmt.Fscan(in, &n)

	ans, cnt := 0, 0
	for i := 0; i < n; i++ {
		var v int
		fmt.Fscan(in, &v)

		if cnt == 0 {
			ans = v
		}

		if v == ans {
			cnt++
		} else {
			cnt--
		}
	}

	fmt.Fprintf(out, "%d\n", ans)
}

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	solve(in, out)
}

File: Luogu/ProblemSet/Volume (2300-2399)/Luogu2397/Luogu2397-Solution-01.cpp
#include <bits/stdc++.h>

#define endl "\n"

void solve() {
    int n;
    std::cin >> n;

    int v, ans = 0, cnt = 0;
    for (int i = 0; i < n; i++) {
        std::cin >> v;
        if (cnt == 0) {
            ans = v;
        }

         ans == v ? cnt++ : cnt--;
    }

    std::cout << ans << endl;
}

int main() {
    std::ios::sync_with_stdio(false);

    solve();

    return 0;
}